content/blaze.md

```diff
@@ -0,0 +1,543 @@
+
+Examples below will reference the `listShow` component from the Todos example app.
+
+<h3 id="validate-data-context">Validate data contexts</h3>
+In order to ensure your component is only used in the way you expect, you should validate the data context provided to it. The data context provides the inputs to the component and so should be checked.
+
+XXX: this does not yet work
+
+You can do this in the components `onCreated()` callback, like so:
+
+```js
+Template.listsShow.onCreated(function() {
+  this.autorun(() => {
+    check(Template.data(), new SimpleSchema({
+     list: {blackbox: true},
+     todosReady: {type: Boolean}
+   }));
+  });
+```
+
+By placing the check in an `autorun()` we ensure that even if the data contexts reactively changes, it always fits the expected schema.
+
+<h3 id="name-data-contexts">Name data contexts to template inclusions</h3>
+It's tempting to provide the data context of a sub-template as a "raw" object (like `{{> todosItem todo}}`), it's a better idea to explicitly give it a name (`{{> todosItem todo=todo}}`). There are two primary reasons for this:
+
+1. When using the data in the sub-template, it's a lot clearer what you are accessing `{{todo.title}}` is clearer than `title`.
+2. It's more flexible, in case in future you need to provide more arguments to the template.
+
+  For instance, in the case of the `todosItem` sub-template, we need to provide two extra arguments to control the editing state of the item, which would have been a hassle to add if the item was used with a raw `todo` argument.
+
+Additionally, for similar reasons of clarity, always explicitly provide a data context to an inclusion, rather than letting it passively "fall-through".
+
+<h3 id="use-each-in">Prefer `{{#each .. in}}`</h3>
+For similar reasons to the above, it's better to use `{{#each todo in todos}}` rather than the older `{{#each todos}}`. The second sets the data context of it's internals to a raw `todo`, and makes it difficult to access the containing template's other data context. 
+
+The only reason not to use `{{#each .. in}}` because it makes it difficult to access the `todo` symbol inside event handlers. Typically the solution to this is simply to use a sub-component to render the inside of the loop.
+
+<h3 id="use-template-instance">Use the template instance</h3>
+Although Blaze doesn't currently have a fully baked component model, you can use the *template instance* as a convenient place to modularize functionality. The template instance is `this` inside template lifecycle callbacks and can be accessed in event handlers and helpers as `Template.instance()`. It's also passed as second argument to event handlers.
```

Comment:
"Although Blaze doesn't currently have a fully baked component model" is overly negative IMO. Plus I bet certain people would disagree with this statement.